backend/open_webui/custom_functions/pipes/albert_rag/albert_rag.py

```diff
@@ -0,0 +1,482 @@
+"""
+title: Albert Rag
+author: Camille Andre
+version: 0.1
+"""
+
+import re
+import requests
+from typing import List, Optional, Dict
+from open_webui.config import OPENAI_API_KEYS, OPENAI_API_BASE_URL
+from openai import OpenAI
+
+
+### Parenthesis for confidence
+async def confidence_message(client, model, context, question, answer, __event_emitter__):
```

Comment:
You might want to try a more robust (and computationally less expensive) hallucination detection framework like [LettuceDetect](https://github.com/KRLabsOrg/LettuceDetect) in the future.